lib/rules/vars-on-top.js

```diff
@@ -103,7 +105,22 @@ module.exports = function(context) {
             var parent = ancestors.pop();
             var grandParent = ancestors.pop();
 
-            if (node.kind === "var") { // check variable is `var` type and not `let` or `const`
+            // check variable is `var` type and not `let` or `const`
+            if (node.kind === "var" && parent.type !== "ExportNamedDeclaration") {
+                if (parent.type === "Program") { // That means its a global variable
+                    globalVarCheck(node, parent);
+                } else {
+                    blockScopeVarCheck(node, parent, grandParent);
+                }
+            }
+        },
+        "ExportNamedDeclaration": function(node) {
```

Comment:
The same basic logic is repeated for ExportNamedDeclaration and VariableDeclaration. You should be able to use just VariableDeclaration and check if the parent is ExportNamedDeclaration.